src/sentry/static/sentry/app/views/projectFilters/projectFilters.jsx

```diff
@@ -0,0 +1,404 @@
+    };
+  },
+
+  formatData(rawData) {
+    let cutOverDate = moment([2017, 8, 11]); // date when detailed stats started being recorded
+
+    return Object.keys(this.getStatOpts()).map(stat => {
+      return {
+        data: rawData[stat].map(([x, y]) => {
+          if (y > 0) {
+            this.setState({blankStats: false});
+          }
+
+          //TODO(maxbittker) this is only needed until October 10th, 2017 :
+          let statDate = moment(x * 1000);
+          let timeSince = cutOverDate.diff(statDate, 'days');
+          // this means detailed stats are available
+          if (
+            (timeSince < 0 && stat === 'blacklisted') ||
+            (timeSince >= 0 && stat !== 'blacklisted')
+          ) {
+            return {x, y: 0};
+          }
+          //END
+
+          return {x, y};
+        }),
+        label: this.getStatOpts()[stat],
+        statName: stat
+      };
+    });
+  },
+
+  getFilterStats() {
+    let statOptions = Object.keys(this.getStatOpts());
+    let {orgId, projectId} = this.props.params;
+    let statEndpoint = `/projects/${orgId}/${projectId}/stats/`;
+    let query = {
+      since: this.state.querySince,
+      until: this.state.queryUntil,
+      resolution: '1d'
+    };
+    $.when
+      .apply(
+        $,
+        // parallelize requests for each statistic
+        statOptions.map(stat => {
+          let deferred = $.Deferred();
+          this.api.request(statEndpoint, {
+            query: Object.assign({stat}, query),
+            success: deferred.resolve.bind(deferred),
+            error: deferred.reject.bind(deferred)
+          });
+          return deferred;
+        })
+      )
+      .done(
+        function(/* statOption1, statOption2, ... statOptionN */) {
+          let rawStatsData = {};
+          let expected = this.state.expected - 1;
+          // when there is a single request made, this is inexplicably called without being wrapped in an array
+          if (statOptions.length === 1) {
+            rawStatsData[statOptions[0]] = arguments[0];
+          } else {
+            for (let i = 0; i < statOptions.length; i++) {
+              rawStatsData[statOptions[i]] = arguments[i][0];
+            }
+          }
+
+          this.setState({
+            rawStatsData,
+            formattedData: this.formatData(rawStatsData),
+            expected,
+            loading: expected > 0
+          });
+        }.bind(this)
+      )
+      .fail(
+        function() {
+          this.setState({error: true});
+        }.bind(this)
+      );
+  },
+
+  fetchData() {
+    let {orgId, projectId} = this.props.params;
+    this.api.request(`/projects/${orgId}/${projectId}/filters/`, {
+      success: (data, textStatus, jqXHR) => {
+        this.setState({filterList: data});
+      },
+      error: () => {
+        this.setState({error: true});
+      },
+      complete: () => {
+        let expected = this.state.expected - 1;
```

Comment:
`expected` would be better out of react state, not a big deal here, I just generally try to keep things out react state.

These API calls would be better served as promises (same with `this.getFilterStats()`) instead of trying to manage a count of api responses in react state, it's less error prone to use `Promise.all()`.

Another thing to keep in mind is that `expected` is never reset to initial value when `fetchData` is called (this could be desired behavior).